### DISCUSSION TITLE
Is there a place where we can see the development roadmap?

### DISCUSSION THREAD

**Discussion by emummel20** [uiua-lang/uiua#292] [General]
First off, can I quickly just say, I am SO excited for Uiua. I'm giddy :D My props to Kai for conceiving it.

I'm coming from Prolog, and if you check out the [Scryer github repo](https://github.com/mthom/scryer-prolog) they have an actual roadmap checklist in the readme that they're working through which includes things like, you know, major predicates to be implemented for the std lib, major performance milestones they're looking to meet, research papers they're basing their work on, major features, etc.

Anything like that we could take a look at for Uiua?

**Comment by kaikalii** [comment]
I don't have anything like that right now.
I'd have to spend some time collecting a lot of thoughts and planning, which is not really how I've been approaching the development thus far.

**Reply by logik-games** [reply]
Hi @kaikalii , mind a quick follow up on this?

Regarding the instability note on the site, I was wondering if there is any eta for when you think stability will be reached? (**no rush at all**)

The reason I ask is because I am considering the tech stack for a long term personal project and for the programming lang of choice I'm down to uiua and prolog. I would _love_ to choose uiua.

Can I also ask: You're in this for the long haul, correct? Uiua is not just an intellectual exercise for you? The intention is to fully develop it into a viable production language, feature and performance-wise?

Thanks!!

**Reply by kaikalii** [reply]
Uiua will be *declared* stable after it has be functionally stable for some time, likely at least a period of a few months.
If I were to estimate, I would expect this to be the case somewhere between 9 months and 2 years from now.

At the moment, I would only choose Uiua for a bigger project if you are okay with eating the cost of having to update code when a breaking change occurs, pre-stability. You also have to be okay with the fact of the various bugs you may encounter. For this reason, unless you just *really* enjoy the language enough for those to be worth it, I would discourage using Uiua in bigger, long-term projects for the time being. And for the love of god don't use it for anything important for business or safety!
While the whole foundation of the language is unlikely to be ripped out from under you, various smaller, nuanced behaviors are quite subject to change.
Any change that I make that I can provide automatic backwards compatibility for, I try to do for as long as possible. Think things like glyph changes or function deprecation.
For a list of the kinds of changes, breaking or otherwise, that occur for each release in Uiua's current unstable state, see [the changelog](https://www.uiua.org/docs/changelog).

And to answer your last question, yes, I am in this for the long haul. My end goal for Uiua is to be able to use it for the majority of my own personal programming projects, perhaps even my professional ones.

Thanks for your interest. Let me know if you have any other questions.
